tests/test_base_models.py

```diff
@@ -241,6 +241,23 @@ def test_save_load(self):
         for i, prediction in enumerate(predictions):
             self.assertEqual(prediction, new_predictions[i])
 
+    def test_fp_16_equality(self):
+        config = self.default_config(save_adam_vars=False, float_16_predict=True)
+        model = Classifier(**config)
+        train_sample = self.dataset.sample(n=self.n_sample)
+        valid_sample = self.dataset.sample(n=self.n_sample)
+        model.fit(train_sample.Text, train_sample.Target)
+        predictions = model.predict(valid_sample.Text)
+        model.config.float_16_predict = False
+        # important to do it this way for the models that do not support fp16 as
```

Comment:
Don't quite follow this comment. Could you explain further?

Reply:
So when we explicitly set model.config.* it doesn't go through the config resolution step like if you set the config items properly. Only BERT and RoBERTa support the float_16 predict but in these cases a warning is raised and this config argument is set back to False. If we start with False and then use `model.config.float_16_predict = True` on any of the base models where it is not supported this will fail because the config resolution doesn't catch it.

Reply:
So the real question I guess is why do we test the models that don't support it yet, but I think it's important to get defined behaviour in these cases so that someone with a GPU that supports it can just set this config argument for every model and not worry about getting undefined behaviour.

Another option would be to explicitly raise an exception in these cases and then change the tests accordingly.

We should maybe think about moving the config resolution into the config object itself though so that any time a conflicting set of config arguments are set this is resolved no matter how or when they are set.

Reply:
yeah, moving to the config object seems like the cleanest approach.